htlcswitch/link.go

```diff
@@ -1652,6 +1653,17 @@ func (l *channelLink) handleDownstreamUpdateAdd(pkt *htlcPacket) error {
 		)
 	}
 
+	// If the channel is quiescent then we issue a temporary channel failure
+	// and bounce it.
+	if !l.quiescer.canSendUpdates() {
```

Comment:
Same above re folding into `IsDraining`.

Reply:
I wonder also if we can fashion things s.t underneath everything uses the `quiescer`, but depending on the input signal/event, it may or may not send/expect the `stfu` message.

Reply:
I think your other idea of having a more flexible TrafficControl interface is the better route. Quiescence is all or nothing. Flushing is about preventing adds only. If you want to consolidate we should choose a flexible core and then we can add some convenience language over the top of it.

Reply:
Looks like we can also now combine the body of this, with the call to `IsFlushing`, so `IsFlushing || !l.quiescer.canSendUpdates() {`